### PR TITLE
<Calendar/> - remove shadow and border-radius from component

### PR DESCRIPTION
**Pull request by yardnsm**
This PR removes the `box-shadow` and `border-raduis` properties from the `<Calendar/>` component. The container component (`<DatePicker/>`, `<Tooltip/>`) is responsible for setting these.

This PR uses the feature from https://github.com/wix/wix-style-react/pull/2200.

Resolves https://github.com/wix-private/wsr-issues/issues/246.

### CODE REVIEW

File: src/DatePicker/DatePicker.scss [hunk 1]
```diff
@@ -8,6 +8,11 @@
   z-index: 1;
 }
 
+.calendarRoot :global .DayPicker {
```
Comment (shlomitc) [inline src/DatePicker/DatePicker.scss]: that doesn't seem correct.
are you sure that `DayPicker` should be customized?
Why can't it be just a property of `calendarRoot`?

File: stories/Calendar/ExampleTooltip.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import Calendar from '../../src/Calendar';
```
Comment (shlomitc) [inline stories/Calendar/ExampleTooltip.js]: all these imports should be to `'wix-style-react/Calendar'`, as this is a snippers that users will use

File: src/DatePicker/DatePicker.scss (hunk 1 above)
Comment (yardnsm) [reply src/DatePicker/DatePicker.scss]: All of the styles in `<Calendar/>` applied to the `.DatePicker` class. If we want to have the border radius and the shadow correctly, we have to set these styles to `.DatePicker`.

Comment (yardnsm) [reply src/DatePicker/DatePicker.scss]: It is. We can set these styles for `.calendarRoot`, but we'll need to also set `overflow: hidden`.

**Comment by milkyfruit** [comment]
@yardnsm calendar in `<DatePicker/>` and `<Range />` looks great.
`<Calendar/>` component also looks correct, only this example isn't right:
screen shot 2018-09-10 at 16 37 04
When a calendar appears inside a popover, popover's 8px border radius isn't visible. Adding `overflow: hidden` fixes the issue.

**Comment by Faradey27** [comment]
@milkyfruit fixed

File: src/Calendar/Calendar.js [hunk 3]
```diff
@@ -136,7 +137,7 @@ export default class Calendar extends WixComponent {
 
   render() {
     return (
-      <div className={styles.calendar}>
+      <div className={`${styles.calendar} ${this.props.className}`}>
```
Comment (shlomitc) [inline src/Calendar/Calendar.js]: use `classnames` package here

File: src/Tooltip/Tooltip.protractor.driver.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
 const tooltipDriverFactory = component => ({
   clickButton: () => component.click(),
+  clickOn: datahook => component.$(`[data-hook="${datahook}"]`).click(),
```
Comment (shlomitc) [inline src/Tooltip/Tooltip.protractor.driver.js]: I don't really like the freedom done here. if the consumer has the `data-hook` in this section, it means he knows the component he's clicking. For example, if it's a button, he'll use the Button testkit

File: src/Tooltip/Tooltip.protractor.driver.js [hunk 5]
```diff
@@ -1,5 +1,7 @@
...
+  togglePopover: () => component.$('[data-hook="popover-button"]').click(),
```
Comment (shlomitc) [inline src/Tooltip/Tooltip.protractor.driver.js]: why not just `click`?

File: src/Tooltip/Tooltip.e2e.js [hunk 6]
```diff
@@ -37,6 +37,19 @@ describe('Tooltip', () => {
...
     });
+
+    eyes.it('should show tooltip in popover mode', async () => {
+      const EC = protractor.ExpectedConditions;
+      const driver = tooltipTestkitFactory({dataHook});
+      await driver.togglePopover();
+      eyes.checkWindow('Check tooltip with popover styles');
+      await waitForVisibilityOf(driver.getTooltipContentElement(tooltipDataHook), 'Cannot find Tooltip content');
+      await driver.clickOn('tooltip-anchor');
+      await browser.wait(EC.not(EC.presenceOf(driver.getTooltipContentElement(tooltipDataHook))));
```
Comment (shlomitc) [inline src/Tooltip/Tooltip.e2e.js]: can't you use `waitForVisibilityOf` and negate the result?

File: src/Tooltip/Tooltip.e2e.js [hunk 7]
```diff
@@ -37,6 +37,19 @@ describe('Tooltip', () => {
...
         });
     });
+
+    eyes.it('should show tooltip in popover mode', async () => {
+      const EC = protractor.ExpectedConditions;
+      const driver = tooltipTestkitFactory({dataHook});
+      await driver.togglePopover();
+      eyes.checkWindow('Check tooltip with popover styles');
+      await waitForVisibilityOf(driver.getTooltipContentElement(tooltipDataHook), 'Cannot find Tooltip content');
+      await driver.clickOn('tooltip-anchor');
```
Comment (shlomitc) [inline src/Tooltip/Tooltip.e2e.js]: see comment in the driver

File: src/DatePicker/DatePicker.scss [hunk 8]
```diff
@@ -4,8 +4,12 @@
   display: inline-block;
 }
 
-.calendarRoot {
+.date-picker-calendar {
   z-index: 1;
+  :global .DayPicker {
+    box-shadow: $Shadow30;
```
Comment (shlomitc) [inline src/DatePicker/DatePicker.scss]: can this be shared across Tooltip and DatePicker? looks like a common css.

File: src/Tooltip/Tooltip.js [hunk 9]
```diff
@@ -13,6 +13,13 @@ const renderSubtreeIntoContainer = ReactDOM.unstable_renderSubtreeIntoContainer;
 //maintain a 60fps rendering
 const createAThrottledOptimizedFunction = cb => () => window.requestAnimationFrame(throttle(cb, 16));
 
+const popoverConfig = {
```
Comment (shlomitc) [inline src/Tooltip/Tooltip.js]: `theme` should be `light` by default

File: stories/Calendar/ExampleTooltip.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  <Tooltip
+    popover
+    placement="bottom"
+    alignment="left"
+    showArrow={false}
+    hideTrigger="click"
+    shouldCloseOnClickOutside
+    maxWidth="300px"
+    content={(
+      <Calendar value={new Date()} onChange={() => {}}/>
```
Comment (shlomitc) [inline stories/Calendar/ExampleTooltip.js]: image

is it possible to fix in this PR as well or should be a separate one?
https://github.com/wix/wix-style-react/issues/2213

File: src/Tooltip/Tooltip.js [hunk 11]
```diff
@@ -13,6 +13,13 @@ const renderSubtreeIntoContainer = ReactDOM.unstable_renderSubtreeIntoContainer;
...
+  padding: 0,
+  overflow: 'hidden',
+  showTrigger: 'click',
+  hideTrigger: 'click'
+};
```
Comment (shlomitc) [inline src/Tooltip/Tooltip.js]: the arrow should be lower by default as it causes it to overlap,
image

File: src/Tooltip/Tooltip.js (hunk 9 above)
Comment (Faradey27) [reply src/Tooltip/Tooltip.js]: not understood

Comment (shlomitc) [reply src/Tooltip/Tooltip.js]: tooltip sets the default `theme` prop to `dark` (see in the story).
for popover - it should be the `light` one

File: src/Tooltip/Tooltip.e2e.js (hunk 6 above)
Comment (Faradey27) [reply src/Tooltip/Tooltip.e2e.js]: i do not like idea to try,catch waitForVisibilityOf + see an error message inside of the logs

File: stories/Calendar/ExampleTooltip.js (hunk 10 above)
Comment (Faradey27) [reply stories/Calendar/ExampleTooltip.js]: will fix

File: src/Tooltip/Tooltip.js (hunk 11 above)
Comment (Faradey27) [reply src/Tooltip/Tooltip.js]: position is correct, the problem is in overflow(and zIndex), i ll hide arrow for popover

File: src/Tooltip/Tooltip.js (hunk 9 above)
Comment (Faradey27) [reply src/Tooltip/Tooltip.js]: done

File: stories/Calendar/ExampleTooltip.js (hunk 10 above)
Comment (Faradey27) [reply stories/Calendar/ExampleTooltip.js]: fixed

**Comment by Faradey27** [comment]
@shlomitc all comments are fixed or commented

File: common.scss [hunk 12]
```diff
@@ -0,0 +1,45 @@
+@import './common/Focusable/FocusableMixins.scss';
+@import './colors.scss';
+
+@mixin popoverTheme {
```
Comment (shlomitc) [inline common.scss]: I this it should sit in a separate area, maybe next to the Tooltip.
having it here will make it impossible to find

File: src/Tooltip/Tooltip.driver.js [hunk 13]
```diff
@@ -86,6 +86,11 @@ const tooltipDriverFactory = ({element, wrapper}) => {
       const values = content.style._values;
       return values.padding;
     },
+    getOverflow: () => {
```
Comment (shlomitc) [inline src/Tooltip/Tooltip.driver.js]: I don't think we should have this kind of function in the driver.
I prefer getting the element instead and verify it in the test.
This is not a function consumers will use in their projects

File: src/Tooltip/Tooltip.js (hunk 11 above)
Comment (shlomitc) [reply src/Tooltip/Tooltip.js]: the arrow direction is wrong. it points inside the tooltip instead of outside

File: src/Tooltip/Tooltip.protractor.driver.js [hunk 14]
```diff
@@ -1,5 +1,7 @@
 const tooltipDriverFactory = component => ({
   clickButton: () => component.click(),
+  clickOnTooltipAnchor: (datahook = 'tooltip-anchor') => component.$(`[data-hook="${datahook}"]`).click(),
```
Comment (shlomitc) [inline src/Tooltip/Tooltip.protractor.driver.js]: this is not part of the component, you can not expose it in the driver

File: src/Tooltip/Tooltip.protractor.driver.js [hunk 15]
```diff
@@ -1,5 +1,7 @@
...
+  click: () => component.$('[data-hook="popover-button"]').click(),
```
Comment (shlomitc) [inline src/Tooltip/Tooltip.protractor.driver.js]: same here

File: src/Tooltip/Tooltip.spec.js [hunk 16]
```diff
@@ -366,6 +366,39 @@ describe('Tooltip', () => {
...
+      return resolveIn(30).then(() => {
+        expect(driver.isShown()).toBeTruthy();
+        driver.click();
+        return resolveIn(30).then(() => {
+          expect(driver.isShown()).toBeFalsy();
+        });
+      });
+    });
+
+    it('should not have a padding and should have overflow: hidden', () => {
```
Comment (shlomitc) [inline src/Tooltip/Tooltip.spec.js]: just call it - it should display the right theme.
This is checked in the eyes section but it's ok if it's here as well

File: src/Tooltip/TooltipContent.js [hunk 17]
```diff
@@ -10,6 +10,9 @@ class TooltipContent extends Component {
     /** alignment of the tooltip's text  */
     textAlign: PropTypes.string,
 
+    /** The tooltip overflow */
+    overflow: PropTypes.string,
```
Comment (shlomitc) [inline src/Tooltip/TooltipContent.js]: why is this necessary?

File: typography.scss [hunk 18]
```diff
@@ -0,0 +1,140 @@
+/*
```
Comment (shlomitc) [inline typography.scss]: I think this file shouldn't appear here

File: src/Calendar/Calendar.scss [hunk 19]
```diff
@@ -50,6 +48,7 @@ $hoverBoxShadow: 0 0 0 $boxShadowWidth $F00;
     width: 17px;
     padding: 18px 0 12px;
     color: $D10;
+    word-wrap: normal;
```
Comment (shlomitc) [inline src/Calendar/Calendar.scss]: is this necessary here by default or only in the case the Tooltip is wrapping it?

Comment (Faradey27) [reply src/Calendar/Calendar.scss]: for me, it should be default

File: src/Tooltip/TooltipContent.js (hunk 17 above)
Comment (Faradey27) [reply src/Tooltip/TooltipContent.js]: how would i pass overflow inside?

File: common.scss (hunk 12 above)
Comment (Faradey27) [reply common.scss]: it used inside of the Tooltip and DatePicker, for me it should not be inside of the Tooltip

File: src/Tooltip/Tooltip.driver.js (hunk 13 above)
Comment (Faradey27) [reply src/Tooltip/Tooltip.driver.js]: you want same logic just inside of the test?

File: common.scss (hunk 12 above)
Comment (shlomitc) [reply common.scss]: you're right it shouldn't sit as part of the tooltip scss file, it should be in common section.
however - this file is a bad mixture, it doesn't have any meaning. colors, focus, fonts. 
so I think we should have a new scss file inside the Tooltip folder that shares this common behavior.
it's then easy to understand where styles should come from and for what purpose

File: src/Tooltip/Tooltip.driver.js (hunk 13 above)
Comment (shlomitc) [reply src/Tooltip/Tooltip.driver.js]: yes

File: src/Tooltip/Tooltip.protractor.driver.js (hunk 14 above)
Comment (shlomitc) [reply src/Tooltip/Tooltip.protractor.driver.js]: need to trigger the tooltip. if it comes as part of the tooltip, awesome. if not, find the relevant triggering element (button?) and click it

File: src/Tooltip/TooltipContent.js (hunk 17 above)
Comment (shlomitc) [reply src/Tooltip/TooltipContent.js]: why do you need the overflow at all?

File: typography.scss (hunk 18 above)
Comment (shlomitc) [reply typography.scss]: rebase shouldn't do that. looks to me like this commit pushed it inside
https://github.com/wix/wix-style-react/pull/2205/commits/edbba59eb97578d57f4f8cc087aa0f9da85ed026#diff-35d7f79fc5e005a5d077c11beac4ece9

File: src/Tooltip/Tooltip.e2e.js (hunk 6 above)
Comment (shlomitc) [reply src/Tooltip/Tooltip.e2e.js]: ok, keep it

File: src/Calendar/Calendar.scss (hunk 19 above)
Comment (shlomitc) [reply src/Calendar/Calendar.scss]: This is the default behavior in css, but the fact that you need to re-decalre it is because the Tooltip sets it differently

I think this css rule should be set only for tooltips and not popovers
https://github.com/wix/wix-style-react/blob/master/src/Tooltip/TooltipContent.scss#L20

File: src/Tooltip/TooltipContent.js (hunk 17 above)
Comment (Faradey27) [reply src/Tooltip/TooltipContent.js]: we disscuss this already, i need it to show correct border, see Domas comments

File: typography.scss (hunk 18 above)
Comment (Faradey27) [reply typography.scss]: this file not visible on my machine, i think problem in case-sensetive
